src/components/BlogCreation.stories.js

```diff
@@ -1,18 +1,22 @@
 import NewPost from './BlogCreation';
 
-// import { action } from '@storybook/addon-actions';
+import { action } from '@storybook/addon-actions';
 
 export default {
   title: 'Post',
   component: NewPost,
   excludeStories: /.*Data$/,
 };
 
+export const actionsData = {
+    uploadPost: action('upload-post'),
+  };
+
 const Template = (args, { argTypes }) => ({
   components: { NewPost },
   props: Object.keys(argTypes),
-//   methods: actionsData,
-  template: '<NewPost v-bind="$props" />',
+  methods: actionsData,
+  template: '<NewPost v-bind="$props @upload-post="uploadPost"" />',
```

Comment:
@ricomadeit this might cause an issue. Try `v-bind="$props" @upload-post="uploadPost"`